plugins/www/Optimove.js

```diff
@@ -1,44 +1,204 @@
+var exec = require("cordova/exec");
 
+var inAppInboxUpdatedHandler;
 
-var exec = require('cordova/exec');
+var currentConfig = {
+  pushReceivedHandler: pushReceivedHandler, //function that receives one argument, a push message object
```

Comment:
What are these implementations currently set to, I don't see `pushReceivedHandler` etc. definitions.

Perhaps we declare some 'no-op' function `function noop() {}` and set a reference to it for all of these by default.

What do you think?

Reply:
(alternately setting them explicitly to undefined or null is fine)

Reply:
I think the noop is a good option, I'll add it now

Reply:
maybe we could give a different no op implementation for each function depending on its parameters? or it's good enough one no op function for all?

Reply:
why noop is better than null? If consumers decide to unset a handler they are likely to set to null, not to empty function. Then we need to handle it being null anyway/

for example here
```
  if (typeof currentConfig[handlerName] === "function") {
    currentConfig[handlerName](message.data);
  } else {
    console.log(`Optimove: No handler defined for '${message.type}' event`);
  }
```
with noop we will never log 'handler not defined'